src/Php74/Php74.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Polyfill\Php74;
+
+/**
+ * @author Ion Bazan <ion.bazan@gmail.com>
+ *
+ * @internal
+ */
+final class Php74
+{
+    /**
+     * @param object $obj
+     *
+     * @return array
+     */
+    public static function get_mangled_object_vars($obj)
+    {
+        $result = array();
+
+        if ($obj instanceof \ArrayIterator || $obj instanceof \ArrayObject) {
+            $reflector = new \ReflectionClass($obj instanceof \ArrayIterator ? 'ArrayIterator' : 'ArrayObject');
+            $flags = $reflector->getMethod('getFlags')->invoke($obj);
+            $reflector = $reflector->getMethod('setFlags');
+
+            $reflector->invoke($obj, ($flags & \ArrayObject::STD_PROP_LIST) ? 0 : \ArrayObject::STD_PROP_LIST);
+            $result = (array) $obj;
+            $reflector->invoke($obj, $flags);
+
+            return $result;
+        }
+
+        foreach ((array) $obj as $key => $value) {
```

Comment:
This breaks references + should be fixed when using ArrayIterator/ArrayObject too.
But do we care really? That's mostly a theoretical issue to me, better ignore?

Reply:
Do you suggest `foreach ($arr as $key => &$value)`?

Reply:
do we need the foreach at all ?